src/server/bg_services/log_replication_scanner.js

```diff
@@ -107,19 +109,30 @@ class LogReplicationScanner {
      * @param {*} src_bucket source bucket
      * @param {*} dst_bucket destination bucket
      * @param {nb.ReplicationLogCandidates} candidates 
+     * @param {boolean} sync_versions should we sync object versions
      */
-    async process_candidates(src_bucket, dst_bucket, candidates) {
+    async process_candidates(src_bucket, dst_bucket, candidates, sync_versions) {
+        return sync_versions ? this.process_candidates_sync_version(src_bucket, dst_bucket, candidates) :
+            this.process_candidates_not_sync_version(src_bucket, dst_bucket, candidates);
+    }
+
+    async process_candidates_sync_version(src_bucket, dst_bucket, candidates) {
+        dbg.warn('process_candidates_sync_version is not yet supported, replicating without sync_version');
+        return this.process_candidates_not_sync_version(src_bucket, dst_bucket, candidates);
+    }
+
+    async process_candidates_not_sync_version(src_bucket, dst_bucket, candidates) {
         const src_dst_objects_list = await this.head_objects(src_bucket.name, dst_bucket.name, candidates);
 
         dbg.log1('log_replication_scanner: process_candidates src_dst_objects_list: ', src_dst_objects_list);
 
-        const copy_keys = [];
+        let copy_keys = {};
         const delete_keys = [];
 
         for (const candidate of src_dst_objects_list) {
             const action = this.process_candidate(candidate);
             if (action === 'copy') {
-                copy_keys.push(candidate.key);
+                copy_keys = { ...copy_keys, [candidate.key]: candidate.src_object_info };
```

Comment:
why not keeping it const and just: 
```suggestion
                copy_keys[candidate.key] = candidate.src_object_info;
```